### ISSUE TITLE
Simple Black-Scholes calculation inconsistency in RQuantLib

### ISSUE THREAD

**Issue by wwyws0000** [eddelbuettel/rquantlib#103]
Hi, 

I'm using the RQuantLib package to do simple options pricing. When comparing results between using EuropeanOption() and a simple, personally written pricing function, I noticed that the results are the same when time to maturity is > 0.1, but differs slightly when time to maturity is < 0.1.  Could you please clarify why this would occur? Is the calculation methodology different from the standard Black-Scholes formula when time to maturity <0.1 in RQuantLib?

Please find below the example:
#Simple Black-Scholes Calculator for call options
```RCode
simpleBSCalc <- function(S,K,D,Rf,t,sigma){
  
  d1 <- (log(S/K) + (Rf - D + sigma^2/2) * t)/(sigma * sqrt(t))
  d2 <- d1 - sigma * sqrt(t)
  
  
  price <- S*exp(-D*t)*pnorm(d1) - K * exp(-Rf*t)*pnorm(d2)
  delta <- pnorm(d1)*exp(-D*t)
  theta <- -((S*sigma * exp(-D*t)/(2*sqrt(t))) * (exp(-d1^2/2)/sqrt(2*pi))) - Rf*K*exp(-Rf*t)*pnorm(d2) + D*S*exp(-D*t)*pnorm(d1)
  
  
  gamma <- exp(-D*t)/(S*sigma*sqrt(t)) * (exp(-d1^2/2)/sqrt(2*pi))
  vega <- S*exp(-D*t)*sqrt(t)*(exp(-d1^2/2)/sqrt(2*pi))
  
  
  out <- c(price = price,delta = delta, gamma = gamma, vega = vega, theta = theta)
  out
}
```

#Compare results when time to maturity is 0.2 (>0.1). The output is the same
```RCode
ttm <- 0.2
EuropeanOption("call", underlying =2000, strike =2000, dividendYield = 0.03, riskFreeRate=0.04, maturity = ttm, volatility = 0.1)
simpleBSCalc(2000,2000,0.03,0.04,ttm,0.1)
```

#Compare results when time to maturity is 0.09 (<0.1), the output slightly differs
```RCode
ttm <- 0.09
EuropeanOption("call", underlying =2000, strike =2000, dividendYield = 0.03, riskFreeRate=0.04, maturity = ttm, volatility = 0.1)
simpleBSCalc(2000,2000,0.03,0.04,ttm,0.1)
```

**Comment by eddelbuettel** [comment]
Can you take a look at the link on each page labeled [Styling with Markdown is supported](https://guides.github.com/features/mastering-markdown/) (or follow this link) and edit your post?  You want triple backticks for code.  That would make it a little more readable,  Thanks.

**Comment by wwyws0000** [comment]
Should be fixed. Apologize as I'm new to GitHub and not familiar with the styling. Thanks again.

**Comment by eddelbuettel** [comment]
Sorry to be bossy, but it _really_ is worth learning some markdown which is applicable way beyond GitHub.

As for your issue, I can replicate it. But it is a third or fourth decimal effect, and it goes away once you `ttm` gets smaller.  Also not the unusually high spot and strike. 

But I have no idea what causes it, but _RQuantLib_ also does not computation here. We just pass it through to QuantLib.  So you could ask there.  But one thing to keep in mind is the _very_ extended set of unit tests.  See eg [europeanoption.hpp](https://github.com/lballabio/QuantLib/blob/master/test-suite/europeanoption.hpp) and [europeanoption.cpp](https://github.com/lballabio/QuantLib/blob/master/test-suite/europeanoption.cpp).

**Comment by eddelbuettel** [comment]
I'd say it is a non-issue.  Consider the case of `spot=strike=20`:

```r
R> ttm <- 0.09
R> EuropeanOption("call", underlying =20, strike =20, dividendYield = 0.03, 
+                               riskFreeRate=0.04, maturity = ttm, volatility = 0.1)
Concise summary of valuation for EuropeanOption 
  value   delta   gamma    vega   theta     rho  divRho 
 0.2477  0.5165  0.6624  2.3848 -1.4183  0.9075 -0.9298 
R> print(simpleBSCalc(20,20,0.03,0.04,ttm,0.1), digits=4)
  price   delta   gamma    vega   theta 
 0.2477  0.5165  0.6624  2.3848 -1.4183 
R> 
```

**Comment by wwyws0000** [comment]
Thanks Dirk. It appears that my calculation is different from yours with the same inputs somehow:

```RCode
> ttm <- 0.09
> EuropeanOption("call", underlying =20, strike =20, dividendYield = 0.03, 
+                                                 riskFreeRate=0.04, maturity = ttm, volatility = 0.1)
Concise summary of valuation for EuropeanOption 
  value   delta   gamma    vega   theta     rho  divRho 
 0.2461  0.5165  0.6666  2.3701 -1.4266  0.8963 -0.9181 
```
I'm using RQuantLib 0.4.2 and when I load the package it tells me the QuantLib version is 1.6.2 which is older than 1.7. Not sure if this plays a part.

**Comment by eddelbuettel** [comment]
You may need to configure with the intraday option.  See [the README.md](https://github.com/eddelbuettel/rquantlib/blob/master/README.md) for a hint for Windows user, and [this blog post of mine](http://dirk.eddelbuettel.com/blog/2015/12/03/) for context.  The step function shown may be your problem.

```r
R> RQuantLib::getQuantLibCapabilities()
     sessions  intradayDate negativeRates 
        FALSE          TRUE          TRUE 
R> 
```

**Comment by wwyws0000** [comment]
This problem was solved after installing the 0.4.4 version. Thank you again.

**Comment by eddelbuettel** [comment]
Great. You did well in finding that (old) issue.  That requires some careful checking -- this 'integer date' problem is really annoying for small `t`.  

It's too bad CRAN no longer builds the Windows binaries but such is life.
